compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -218,7 +218,12 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
   private def makeParamsString(sym: Symbol, reflProxy: Boolean,
       inRTClass: Boolean): String = {
     val tpe = sym.tpe
-    val paramTypeNames = tpe.params map (p => internalName(p.tpe))
+
+    val paramTypeNames0 = tpe.params map (p => internalName(p.tpe))
+    val paramTypeNames =
+      if (!isScalaJSDefinedJSClass(sym.owner)) paramTypeNames0
+      else internalName(sym.owner.toTypeConstructor) :: paramTypeNames0
+
```

Comment:
This totally overlaps with how the runtime classes are handled (same code is below again). Consider generalizing the `inRTClass` flag and not checking for `isScalaJSDefinedJSClass` in this file. It seems off...